src/System.Private.ServiceModel/tests/Scenarios/Extensibility/MessageInterceptor/MessageInterceptorTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using Infrastructure.Common;
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using Xunit;
+
+public partial class MessageInterceptorTests : ConditionalWcfTest
+{
+    public class MessageModifier : ChannelMessageInterceptor
+    {
+        int send = 0;
+
+        public override void OnSend(ref Message msg)
+        {
+            send++;
+            if (send % 2 == 0)
+            {
+                // Add extra header so that when the message arrives on service side, the message won't be dropped
+                msg.Headers.Add(MessageHeader.CreateHeader("ByPass", "urn:InterceptorNamespace", "ByPassPassword"));
+            }
+        }
+
+        public override ChannelMessageInterceptor Clone()
+        {
+            return new MessageModifier();
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+    [ActiveIssue(1170)]
```

Comment:
It would be good to understand why this test fails before we disable it.   @khdang , can this test be changed to pass, or do we understand the failure enough to know it is a product issue?